File: C++/2013/S2.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int w, n;cin >>w >> n;
    int weights[n + 3];weights[0] = 0;weights[1] = 0;weights[2] = 0;
    bool output = false;

    for (int q=0;q<n;q++) {
        cin >> weights[q + 3];
        if (weights[q] + weights[q + 1] + weights[q + 2] + weights[q + 3] > w) {
            if (not output) {
                cout << q;
                output = true;
            }
        }
    }
    if (not output) cout << n;
}

File: Python/2013/S2.py
weight = int(input())

length = int(input())

weights = [0] * 3 + [int(input()) for _ in range(length)]

output = 'False'
for q in range(len(weights)):
    if sum(weights[q:(q + 4)]) > weight and output == 'False':
        output = q
print(output)